sdcflows/workflows/base.py

```diff
@@ -131,21 +138,54 @@ def init_sdc_estimate_wf(fmaps, epi_meta, omp_nthreads=1, debug=False):
                 'At least one of the EPI runs with alternative phase-encoding '
                 'blips is missing the required "PhaseEncodingDirection" metadata entry.')
 
-        # Find matched PE directions
-        matched_pe = check_pes(fmaps_epi, epi_meta['PhaseEncodingDirection'])
+        # Find matched PE directions. Return pe-dirs for matching/opposing EPIs
+        matched_pe, matched_pe_dir, opposed_pe_dir = check_pes(
+            fmaps_epi, epi_meta['PhaseEncodingDirection'])
 
-        # Get EPI polarities and their metadata
-        sdc_unwarp_wf = init_pepolar_unwarp_wf(
+        # Topup workflow (prepares epis and runs topup)
+        topup_wf = init_topup_wf(
             matched_pe=matched_pe,
             omp_nthreads=omp_nthreads)
-        sdc_unwarp_wf.inputs.inputnode.epi_pe_dir = epi_meta['PhaseEncodingDirection']
-        sdc_unwarp_wf.inputs.inputnode.fmaps_epi = fmaps_epi
+        # set inputs
+        topup_wf.inputs.inputnode.epi_pe_dir = epi_meta['PhaseEncodingDirection']
+        topup_wf.inputs.inputnode.epi_trt = epi_meta['TotalReadoutTime']
+        topup_wf.inputs.inputnode.fmaps_epi = fmaps_epi
+        topup_wf.inputs.inputnode.matched_pe_dir = matched_pe_dir
+        topup_wf.inputs.inputnode.opposed_pe_dir = opposed_pe_dir
+
+        # After topup workflow, recreate FIELDMAP PATH
+        fmap_wf = init_fmap_wf(
+            omp_nthreads=omp_nthreads,
+            fmap_bspline=False)
+
+        fmap2field_wf = init_fmap2field_wf(
+            omp_nthreads=omp_nthreads,
+            debug=debug)
+        # set inputs
+        fmap2field_wf.inputs.inputnode.metadata = epi_meta
+
+        sdc_unwarp_wf = init_sdc_unwarp_wf(
+            omp_nthreads=omp_nthreads,
+            debug=debug)
```

Comment:
This is not a responsibility of this workflow. Instead, the warpfield should be converted to an ITK-type of transform and generate a new (unwarped) reference, as in: https://github.com/nipreps/sdcflows/blob/bf9cd97fa1e0e25d152c02e96120a1ed37b60a54/sdcflows/workflows/pepolar.py#L128-L140

But that's going to be tricky while https://github.com/poldracklab/nitransforms does not support FSL's bspline parametric fields.

Reply:
Not sure what you mean by this not being a responsibility of this workflow. I tried to summarize the approach in the comment to (#106), but essentially the topup-estimated fieldmap is fed into the workflows of the fmap-module.